AppCenterAuth/AppCenterAuth/MSAuth.m

```diff
@@ -181,14 +182,18 @@ - (void)signInInWithCompletionHandler:(MSSignInCompletionHandler _Nullable)compl
   }
   if ([[MS_Reachability reachabilityForInternetConnection] currentReachabilityStatus] == NotReachable) {
     [self callCompletionHandler:completionHandler
-                  withErrorCode:MSACAuthErrorSignInWhenNoConnection
+                  withErrorCode:MSACAuthErrorNoConnection
                      andMessage:@"User sign-in failed. Internet connection is down."];
     return;
   }
   if (self.clientApplication == nil || self.authConfig == nil) {
-    [self callCompletionHandler:completionHandler
-                  withErrorCode:MSACAuthErrorSignInBackgroundOrNotConfigured
-                     andMessage:@"signIn is called while it's not configured or not in the foreground."];
+    if (self.signInShouldWaitForConfig) {
+      MSLogDebug([MSAppCenter logTag], @"Downloading configuration in process. Waiting for it before sign-in.");
```

Comment:
Let's add a comment to explain that when the user try to sign-in before config downloading completes, we only log a Debug message here and let `downloadConfigurationWithETag:eTag` continue the sign-in flow as soon as downloading completes.